Amplify codegen generates wrong return for scalar list (String)

### Before opening, please confirm:


- [X] I have installed the latest version of the Amplify CLI (see above), and confirmed that the issue still persists.
- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-cli/issues?q=is%3Aissue+).
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-cli/blob/master/CONTRIBUTING.md#bug-reports).
- [X] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

### How did you install the Amplify CLI?

curl "https://awscli.amazonaws.com/awscli-exe-linux-x86_64.zip" -o "awscliv2.zip"

### If applicable, what version of Node.js are you using?

v10.19.0

### Amplify CLI Version

5.2.1

### What operating system are you using?

Ubuntu

### Amplify Categories

api

### Amplify Commands

codegen

### Describe the bug

In the appsync schema, I have the following:
```graphql
type Query {
    ...
    getGroups: [String]
    ...
}
```
But when I generate the `api.service.ts` for angular, the function code returns and array of a new type, but this type is not defined nor string:

`Cannot find name 'GetGroupsQuery'.ts(2304)`

```javascript
async GetGroups(): Promise<Array<GetGroupsQuery>> {
  const statement = `query GetGroups {
      getGroups
    }`;
  const response = (await API.graphql(graphqlOperation(statement))) as any;
  return <Array<GetGroupsQuery>>response.data.getGroups;
}
``` 

### Expected behavior

The code generated should return a string array, in fact if change `GetGroupsQuery` for `string`, the app runs without problem

```javascript
async GetGroups(): Promise<Array<string>> {
  const statement = `query GetGroups {
      getGroups
    }`;
  const response = (await API.graphql(graphqlOperation(statement))) as any;
  return <Array<string>>response.data.getGroups;
}
```

### Reproduction steps

1. Create the schema in appsync through AWS Console, with a query that returns an array of strings (Maybe is the same for the other scalars, like numbers, not tested)
```graphql
type Query {
	getGroups: [String]
}
```
2. Add the api to the project as follows `amplify add codegen --apiId xxxxxxxxxxxxxx`
3. Go to the app.service.ts and the linter will show you the problem. The code generated includes an array of a new undefined type instead of scalar type
```javascript
async GetGroups(): Promise<Array<GetGroupsQuery>> {
  const statement = `query GetGroups {
      getGroups
    }`;
  const response = (await API.graphql(graphqlOperation(statement))) as any;
  return <Array<GetGroupsQuery>>response.data.getGroups;
}
```

### GraphQL schema(s)

<details>

```graphql
type Query {
    getGroups: [String]
}
```

</details>


### Log output

<details>

```
Error: src/app/graphql/api.service.ts:123:36 - error TS2304: Cannot find name 'GetGroupsQuery'.

123   async GetGroups(): Promise<Array<GetGroupsQuery>> {
```

</details>


### Additional information

_No response_

--- a/packages/graphql-types-generator/src/angular/index.ts
+++ b/packages/graphql-types-generator/src/angular/index.ts
@@ -1,4 +1,4 @@
-import { GraphQLNonNull, GraphQLType, isScalarType } from 'graphql';
+import { GraphQLNonNull, GraphQLType, isScalarType, isEnumType, isListType } from 'graphql';
 import * as prettier from 'prettier';
 import { LegacyCompilerContext, LegacyOperation, LegacyInlineFragment, LegacyField } from '../compiler/legacyIR';
 
@@ -128,9 +128,19 @@ function getOperationResultField(operation: LegacyOperation): LegacyField | void
 function getReturnTypeName(generator: CodeGenerator, op: LegacyOperation): String {
   const { operationName, operationType } = op;
   const { type } = op.fields[0];
-  if (isScalarType(type)) {
+  //List of scalar or enum type
+  if (isListType(type)) {
+    const { ofType } = type;
+    if (isScalarType(ofType) || isEnumType(ofType)) {
+      return `Array<${typeNameFromGraphQLType(generator.context, ofType)}>`;
+    }
+  }
+  //Scalar and enum type
+  if (isScalarType(type) || isEnumType(type)) {
     return typeNameFromGraphQLType(generator.context, type);
-  } else {
+  }
+  //Non scalar type
+  else {
     let returnType = interfaceNameFromOperation({ operationName, operationType });
     if (op.operationType === 'subscription' && op.fields.length) {
       returnType = `Pick<__SubscriptionContainer, "${op.fields[0].responseName}">`;

--- a/packages/graphql-types-generator/test/angular/index.js
+++ b/packages/graphql-types-generator/test/angular/index.js
@@ -391,5 +391,35 @@ describe('Angular code generation', function() {
       const source = generateSource(context);
       expect(source).toMatchSnapshot();
     });
+
+    test(`should generate simple query operation with scalar/enum field and scalar/enum return type w & w/o list`, function() {
+      const { compileFromSource } = setup(loadSchema(require.resolve('../fixtures/misc/queryWithScalarAndEnumType.graphql')));
+      const context = compileFromSource(`
+      query GetScalars {
+        getScalars
+      }
+      query GetScalar {
+        getScalar
+      }
+      query GetPhones {
+        getPhones {
+          number
+        }
+      }
+      query GetPhone {
+        getPhone {
+          number
+        }
+      }
+      query GetEnums {
+        getEnums
+      }
+      query GetEnum {
+        getEnum
+      }
+      `);
+      const source = generateSource(context);
+      expect(source).toMatchSnapshot();
+    });
   });
 });

--- a/packages/graphql-types-generator/test/angular/index.js
+++ b/packages/graphql-types-generator/test/angular/index.js
@@ -401,16 +401,6 @@ describe('Angular code generation', function() {
       query GetScalar {
         getScalar
       }
-      query GetPhones {
-        getPhones {
-          number
-        }
-      }
-      query GetPhone {
-        getPhone {
-          number
-        }
-      }
       query GetEnums {
         getEnums
       }

--- a/packages/graphql-types-generator/test/fixtures/misc/queryWithScalarAndEnumType.graphql
+++ b/packages/graphql-types-generator/test/fixtures/misc/queryWithScalarAndEnumType.graphql
@@ -2,20 +2,13 @@ type Person {
   id: ID!
   name: String!
   items: [String!]
-  contacts: [Phone!]
   createdAt: AWSDateTime!
   updatedAt: AWSDateTime!
-}
-
-type Phone {
-  number: String!
 }
 
 type Query {
   getScalars: [String]
   getScalar: String
-  getPhones: [Phone]
-  getPhone: Phone
   getEnums: [Status]
   getEnum: Status
   getPerson(id: ID!): Person
@@ -26,10 +19,6 @@ enum Status {
   APPROVED
   PENDING
   REJECTED
-}
-
-input PhoneInput {
-  number: String!
 }
 
 enum ModelSortDirection {
@@ -140,13 +129,11 @@ input CreatePersonInput {
   id: ID
   name: String!
   items: [String!]
-  contacts: [PhoneInput!]
 }
 
 input UpdatePersonInput {
   name: String
   items: [String!]
-  contacts: [PhoneInput!]
 }
 
 input DeletePersonInput {

--- a/packages/graphql-types-generator/test/fixtures/misc/queryWithScalarAndEnumType.graphql
+++ b/packages/graphql-types-generator/test/fixtures/misc/queryWithScalarAndEnumType.graphql
@@ -1,0 +1,174 @@
+type Person {
+  id: ID!
+  name: String!
+  items: [String!]
+  contacts: [Phone!]
+  createdAt: AWSDateTime!
+  updatedAt: AWSDateTime!
+}
+
+type Phone {
+  number: String!
+}
+
+type Query {
+  getScalars: [String]
+  getScalar: String
+  getPhones: [Phone]
+  getPhone: Phone
+  getEnums: [Status]
+  getEnum: Status
+  getPerson(id: ID!): Person
+  listPersons(filter: ModelPersonFilterInput, limit: Int, nextToken: String): ModelPersonConnection
+}
+
+enum Status {
+  APPROVED
+  PENDING
+  REJECTED
+}
+
+input PhoneInput {
+  number: String!
+}
+
+enum ModelSortDirection {
+  ASC
+  DESC
+}
+
+type ModelPersonConnection {
+  items: [Person]
+  nextToken: String
+}
+
+input ModelStringInput {
+  ne: String
+  eq: String
+  le: String
+  lt: String
+  ge: String
+  gt: String
+  contains: String
+  notContains: String
+  between: [String]
+  beginsWith: String
+  attributeExists: Boolean
+  attributeType: ModelAttributeTypes
+  size: ModelSizeInput
+}
+
+input ModelIDInput {
+  ne: ID
+  eq: ID
+  le: ID
+  lt: ID
+  ge: ID
+  gt: ID
+  contains: ID
+  notContains: ID
+  between: [ID]
+  beginsWith: ID
+  attributeExists: Boolean
+  attributeType: ModelAttributeTypes
+  size: ModelSizeInput
+}
+
+input ModelIntInput {
+  ne: Int
+  eq: Int
+  le: Int
+  lt: Int
+  ge: Int
+  gt: Int
+  between: [Int]
+  attributeExists: Boolean
+  attributeType: ModelAttributeTypes
+}
+
+input ModelFloatInput {
+  ne: Float
+  eq: Float
+  le: Float
+  lt: Float
+  ge: Float
+  gt: Float
+  between: [Float]
+  attributeExists: Boolean
+  attributeType: ModelAttributeTypes
+}
+
+input ModelBooleanInput {
+  ne: Boolean
+  eq: Boolean
+  attributeExists: Boolean
+  attributeType: ModelAttributeTypes
+}
+
+input ModelSizeInput {
+  ne: Int
+  eq: Int
+  le: Int
+  lt: Int
+  ge: Int
+  gt: Int
+  between: [Int]
+}
+
+input ModelPersonFilterInput {
+  name: ModelStringInput
+  items: ModelStringInput
+  and: [ModelPersonFilterInput]
+  or: [ModelPersonFilterInput]
+  not: ModelPersonFilterInput
+}
+
+enum ModelAttributeTypes {
+  binary
+  binarySet
+  bool
+  list
+  map
+  number
+  numberSet
+  string
+  stringSet
+  _null
+}
+
+input CreatePersonInput {
+  id: ID
+  name: String!
+  items: [String!]
+  contacts: [PhoneInput!]
+}
+
+input UpdatePersonInput {
+  name: String
+  items: [String!]
+  contacts: [PhoneInput!]
+}
+
+input DeletePersonInput {
+  id: ID!
+}
+
+type Mutation {
+  createPerson(input: CreatePersonInput!, condition: ModelPersonConditionInput): Person
+  updatePerson(input: UpdatePersonInput!, condition: ModelPersonConditionInput): Person
+  deletePerson(input: DeletePersonInput!, condition: ModelPersonConditionInput): Person
+}
+
+input ModelPersonConditionInput {
+  name: ModelStringInput
+  items: ModelStringInput
+  and: [ModelPersonConditionInput]
+  or: [ModelPersonConditionInput]
+  not: ModelPersonConditionInput
+}
+
+type Subscription {
+  onCreatePerson: Person @aws_subscribe(mutations: ["createPerson"])
+  onUpdatePerson: Person @aws_subscribe(mutations: ["updatePerson"])
+  onDeletePerson: Person @aws_subscribe(mutations: ["deletePerson"])
+}

